corehq/apps/hqcase/api/get_list.py

```diff
@@ -76,16 +78,25 @@ def get_list(domain, couch_user, params):
         params_string = b64decode(params['cursor']).decode('utf-8')
         params = QueryDict(params_string, mutable=True)
         # QueryDict.pop() returns a list
+        include_deprecated = params.pop(INCLUDE_DEPRECATED, ['False'])[0].lower().capitalize() == 'True'
         last_date = params.pop(INDEXED_AFTER, [None])[0]
         last_id = params.pop(LAST_CASE_ID, [None])[0]
         query = _get_cursor_query(domain, params, last_date, last_id)
     else:
         params = params.copy()  # Makes params mutable for pagination below
+        include_deprecated = params.pop(INCLUDE_DEPRECATED, ['False'])[0].lower().capitalize() == 'True'
         query = _get_query(domain, params)
 
     if not couch_user.has_permission(domain, 'access_all_locations'):
         query = query_location_restricted_cases(query, domain, couch_user)
 
+    # Cases with deprecated case types should not be returned if the flag is not specified
```

Comment:
I commented to this effect on the spec, but I think it's strange to ignore deprecated types in an API - if I ask an API for a deprecated case type, I'd hope the API assumes that as an advanced user, I know what I'm doing, and it gives me cases of that type (maybe users have old app versions and are still submitting those cases).  You could make a case that the API should throw an exception if I didn't also specify `include_deprecated`, though I don't think that's necessary - I'd guess it depends on how deprecation is used.  As-is, though, this will just return empty results for that case type, meaning that existing integrations will not fail and alert their maintainers, they'll just stop returning data.

Reply:
Should we then rather flip the default of this flag? So, instead of defaulting to not returning cases with deprecated case types, we do. The alternative I had in mind was to simply never apply the `include_deprecated` filter if the user has given a case type in the query params. In this scenario we can assume that the user doesn't care about the deprecated state of that case type, and if the `include_deprecated` parameter is included along with `case_type`, then we simply throw an exception that they can't both be used in conjunction (as it wouldn't really make sense).

@esoergel What are your thoughts on this?

Reply:
You mean like, always include deprecated case types unless the user adds `include_deprecated=False` to the request?  That does seem more explicit, which I like.  What I was suggesting though was to just not add any sort of deprecation based filtering in the API at all.  Or perhaps I'm thinking of this differently than y'all are - could you talk me through the rationale?  Like, why would users want to filter cases by case type deprecation status (whether by default or explicitly) rather than case type?

Relatedly, I expect most of the time users are using this API, they'll be filtering by a single case type.  In fact, I believe @kaapstorm originally proposed adding case type to the URL path so it'd be sort-of a permanent filter.  I didn't go that route, to preserve flexibility (maybe you want to just look up an ID and don't know the type), but I do think users will generally be working with a specific case type, as they need to be able to handle the properties schema associated with it.

Reply:
Exactly, but I have made a commit [here](https://github.com/dimagi/commcare-hq/pull/33089/commits/f23f4eb62edd725b6867a7153fa5ea366db5f50b) which should also address defaulting to including cases with deprecated case types. Alongside making it more explicit, this should also ensure that existing API calls aren't broken for users that don't include the deprecated query param.

I think it's good to have the deprecated case type filter as an optional filter. When a user deprecates a case type, this will usually be related to historic case data. Most of the time users might be filtering by case type, but there could exist times where the case type filter is not used (e.g. when the user wants to retrieve all the cases owned by a specific location). In instances like this, the user might want to exclude historic cases that have a deprecated case type. Ultimately, I think if we support the flexibility of allowing the user to retrieve cases without a case type filter, then we should allow them to filter out cases with deprecated case types as well.

Reply:
thanks, that sounds like a reasonable approach to me